lib/src/main/kotlin/com/swmansion/starknet/data/types/TransactionPayload.kt

```diff
@@ -119,14 +122,10 @@ data class DeclareTransactionV2Payload(
 
     @SerialName("version")
     val version: TransactionVersion,
-
-    @SerialName("type")
-    override val type: TransactionType = TransactionType.DECLARE,
 ) : DeclareTransactionPayload()
 
-@Suppress("DataClassPrivateConstructor")
 @Serializable
-data class DeclareTransactionV3Payload private constructor(
```

Comment:
Why this constructor's no longer private? 

As far as I remember, we kept them private to avoid creation of payloads with invalid values (unsupported, hardcoded v3 fields)

Reply:
Should be private, correct.